### DISCUSSION TITLE
Loader exits with code 1, just using import {FileAttachment}

### DISCUSSION THREAD

**Discussion by Martien** [observablehq/framework#1832] [Q&A] [answered]
I fail trying to create a page loader `pl.md.js` that reads `./src/pl.txt` like so:
```
import { FileAttachment } from "observablehq:stdlib";
const pl = FileAttachment("pl.txt");
```
This gives the error message: `loader exited with code 1`.

Terminal with `npm run dev` shows:
```
Node.js v22.10.0
error in 64ms: loader exited with code 1
Error: loader exited with code 1
    at CommandLoader.exec (file:///Users/martien/Documents/Project/aarde/node_modules/@observablehq/framework/dist/loader.js:430:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///Users/martien/Documents/Project/aarde/node_modules/@observablehq/framework/dist/loader.js:373:11
```

Even having only `import { FileAttachment } from "observablehq:stdlib";` gives a loading error.

Help, please.

(I may have mixed up mental models, as I find imports, files, and page loaders as a whole confusing.)

**Comment by mbostock** [comment, marked as answer]
JavaScript data loaders are simply vanilla Node.js programs. You need to use `node:fs` to read files rather than `FileAttachment`.

**Reply by Martien** [reply]
> JavaScript data loaders are simply vanilla Node.js programs. You need to use `node:fs` to read files rather than `FileAttachment`.

Thanks. Got it working. I’m new to Node.js. Observable has been abstracting it for me so nicely until now.
